measurements/api/measurements.py

```diff
@@ -143,26 +151,40 @@ def get_measurement(measurement_id):
             Autoclaved.filename.label('a_filename'),
             Autoclaved.autoclaved_no.label('a_autoclaved_no'),
 
-    ).filter(Measurement.id == measurement_id)\
+    ).filter(Measurement.msm_no == msm_no)\
         .join(Report, Report.report_no == Measurement.report_no)\
         .join(Autoclaved, Autoclaved.autoclaved_no == Report.autoclaved_no)
     try:
         msmt = q.one()
     except exc.MultipleResultsFound:
         current_app.logger.warning("Duplicate rows for measurement_id: %s" % measurement_id)
         msmt = q.first()
+    except exc.NoResultFound:
+        # XXX we should actually return a 404 here
+        raise BadRequest("No measurement found")
 
     r = requests.get(urljoin(current_app.config['AUTOCLAVED_BASE_URL'], msmt.a_filename),
             headers={"Range": "bytes={}-{}".format(msmt.frame_off, msmt.frame_off+msmt.frame_size)}, stream=True)
```

Comment:
Looks like another off-by-one error, it should be `frame_off + frame_size - 1` as both ends of `Range` query are included.